File: DSA/LinSearch.java
import java.util.*;
import static java.lang.System.*;


class LinSearch
{

/*-----------------Solution Function---------------------*/

 static int linSearch(int []arr,int val)
 {
  int i=-1;
  for(int j=0;j<arr.length;j++)
   if(arr[j]==val)
   {
    i=j;
    break;
   }
  return i;
 }

/*----------------Main Function--------------------------*/

 public static void main(String...arg)
 {
  int arr[]={5,6,2,4,10};

  out.print("Enter Item to find : ");
  Scanner sc=new Scanner(System.in);
  int val=sc.nextInt();

  int i=linSearch(arr,val);
  if(i==-1)
  out.println("Item Not Found..");
  else
  out.printf("Item found at : %d\n",i+1);
 }
}

File: DSA/LinSearch.cpp
// Linear Search

#include<iostream>
using namespace std;

int linSearch(int arr[],int size,int val)
{
 int i=-1;
 for(int j=0;j<size;j++)
  if(arr[j]==val)
  {
   i=j;
   return i;
  }
 return i;
}

int main()
{
 int arr[]={5,6,2,4,10};
 int val;
 cout<<"\nEnter Item to find : ";
 cin>>val;

 int i=LinSearch(arr,5,val);
 if(i==-1)
  cout<<"\nItem Not Found..";
 else
  cout<<"\nItem found at : "<<i+1;
}
